src/test/java/com/github/tomakehurst/wiremock/AdvancedPathPatternSerializationTest.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2024 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingXPath;
+
+import com.github.tomakehurst.wiremock.common.Json;
+import com.github.tomakehurst.wiremock.matching.StringValuePattern;
+import java.util.Map;
+import org.json.JSONException;
+import org.junit.jupiter.api.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+public class AdvancedPathPatternSerializationTest {
+
+  @Test
+  void matchesXpathWithOnlyAValueSerializesCorrectly() throws JSONException {
+    String expectedJson = "{\"matchesXPath\" : \"//AccountId\"}";
+    StringValuePattern pattern = matchingXPath("//AccountId");
+    String serializedPattern = Json.write(pattern);
+    JSONAssert.assertEquals(expectedJson, serializedPattern, true);
+  }
+
+  @Test
+  void matchingXpathWithPatternSerializesCorrectly() throws JSONException {
+    String expectedJson =
+        "{\n"
+            + "        \"matchesXPath\" : {\n"
+            + "            \"expression\" : \"//AccountId\",\n"
+            + "            \"equalTo\" : \"123\"\n"
+            + "        }\n"
+            + "}";
+    StringValuePattern pattern = matchingXPath("//AccountId", equalTo("123"));
+    String serializedPattern = Json.write(pattern);
+    JSONAssert.assertEquals(expectedJson, serializedPattern, true);
```

Comment:
Can I suggest using JSONUnit for these assertions rather than JSONAssert? I think this is the best of the JSON matching libraries we have in play so we should be moving towards using it for everything.

Reply:
Awesome.  I have updated the asserts to use jsonUnit - `assertThat(Json.write(pattern), jsonEquals(expectedJson));`